File: db4onqopt/core/src/com/db4o/nativequery/expr/cmp/ComparisonOperator.java
/* Copyright (C) 2004 - 2006  db4objects Inc.  http://www.db4o.com */

package com.db4o.nativequery.expr.cmp;

// TODO: switch to individual classes and visitor dispatch?
public final class ComparisonOperator {
	public final static int EQUALS_ID=0;
	public final static int SMALLER_ID=1;
	public final static int GREATER_ID=2;
	public final static int CONTAINS_ID=3;
	public final static int STARTSWITH_ID=4;
	public final static int ENDSWITH_ID=5;
	
	public final static ComparisonOperator EQUALS=new ComparisonOperator(EQUALS_ID,"==", true);
	public final static ComparisonOperator SMALLER=new ComparisonOperator(SMALLER_ID,"<", false);
	public final static ComparisonOperator GREATER=new ComparisonOperator(GREATER_ID,">", false);
	public final static ComparisonOperator CONTAINS=new ComparisonOperator(CONTAINS_ID,"<CONTAINS>", false);
	public final static ComparisonOperator STARTSWITH=new ComparisonOperator(STARTSWITH_ID,"<STARTSWITH>", false);
	public final static ComparisonOperator ENDSWITH=new ComparisonOperator(ENDSWITH_ID,"<ENDSWITH>", false);
	
	private int _id;
	private String _op;
	private boolean _symmetric;
	
	private ComparisonOperator(int id, String op, boolean symmetric) {
		_id=id;
		_op=op;
		_symmetric=symmetric;
	}
	
	public int id() {
		return _id;
	}
	
	public String toString() {
		return _op;
	}
	
	public boolean isSymmetric() {
		return _symmetric;
	}
}

File: db4on/core/src/core/com/db4o/nativequery/expr/cmp/ComparisonOperator.cs
namespace com.db4o.nativequery.expr.cmp
{
	public sealed class ComparisonOperator
	{
		public const int EQUALS_ID = 0;

		public const int SMALLER_ID = 1;

		public const int GREATER_ID = 2;

		public const int CONTAINS_ID = 3;

		public const int STARTSWITH_ID = 4;

		public const int ENDSWITH_ID = 5;

		public static readonly com.db4o.nativequery.expr.cmp.ComparisonOperator EQUALS = 
			new com.db4o.nativequery.expr.cmp.ComparisonOperator(EQUALS_ID, "==", true);

		public static readonly com.db4o.nativequery.expr.cmp.ComparisonOperator SMALLER = 
			new com.db4o.nativequery.expr.cmp.ComparisonOperator(SMALLER_ID, "<", false);

		public static readonly com.db4o.nativequery.expr.cmp.ComparisonOperator GREATER = 
			new com.db4o.nativequery.expr.cmp.ComparisonOperator(GREATER_ID, ">", false);

		public static readonly com.db4o.nativequery.expr.cmp.ComparisonOperator CONTAINS = 
			new com.db4o.nativequery.expr.cmp.ComparisonOperator(CONTAINS_ID, "<CONTAINS>", 
			false);

		public static readonly com.db4o.nativequery.expr.cmp.ComparisonOperator STARTSWITH
			 = new com.db4o.nativequery.expr.cmp.ComparisonOperator(STARTSWITH_ID, "<STARTSWITH>"
			, false);

		public static readonly com.db4o.nativequery.expr.cmp.ComparisonOperator ENDSWITH = 
			new com.db4o.nativequery.expr.cmp.ComparisonOperator(ENDSWITH_ID, "<ENDSWITH>", 
			false);

		private int _id;

		private string _op;

		private bool _symmetric;

		private ComparisonOperator(int id, string op, bool symmetric)
		{
			_id = id;
			_op = op;
			_symmetric = symmetric;
		}

		public int Id()
		{
			return _id;
		}

		public override string ToString()
		{
			return _op;
		}

		public bool IsSymmetric()
		{
			return _symmetric;
		}
	}
}
